spec/lib/pre_assembly/batch_spec.rb

```diff
@@ -258,7 +258,7 @@
     it 'returns Array of object_files from all DigitalObjects' do
       fake_files = [[1, 2], [3, 4], [5, 6]]
       fake_dobjs = fake_files.map { |fs| instance_double(PreAssembly::DigitalObject, object_files: fs) }
-      flat_dir_images.digital_objects = fake_dobjs
+      allow(flat_dir_images).to receive(:digital_objects).and_return(fake_dobjs)
```

Comment:
afaict, the `digital_objects` ivar in the current `main` branch only has an `attr_writer` so that this test can set this value?  not totally sure about that, but if so, this is probably a desirable touchup even if we scrap this PR for another approach (would also want to get rid of the `attr_writer` i commented out above).